js/checkboxgroup/checkboxgroup.test.ts

```diff
@@ -0,0 +1,345 @@
+
+		expect(item_one).toBeChecked();
+		expect(item_two).not.toBeChecked();
+	});
+
+	test("renders correct value when passed as number: multiple values", async () => {
+		const { getByLabelText } = await render(CheckboxGroup, {
+			value: [1, 3],
+			label: "Dropdown",
+			choices: [
+				["Choice One", 1],
+				["Choice Two", 2],
+				["Choice Three", 3]
+			]
+		});
+
+		const item_one = getByLabelText("Choice One") as HTMLInputElement;
+		const item_two = getByLabelText("Choice Two") as HTMLInputElement;
+		const item_three = getByLabelText("Choice Three") as HTMLInputElement;
+
+		expect(item_one).toBeChecked();
+		expect(item_two).not.toBeChecked();
+		expect(item_three).toBeChecked();
+	});
+
+	test("component value and rendered value are in sync", async () => {
+		const { getByLabelText, debug, component } = await render(CheckboxGroup, {
+			value: [1, 3],
+			label: "Dropdown",
+			choices: [
+				["Choice One", 1],
+				["Choice Two", 2],
+				["Choice Three", 3]
+			]
+		});
+
+		const item_one = getByLabelText("Choice One") as HTMLInputElement;
+		const item_two = getByLabelText("Choice Two") as HTMLInputElement;
+		const item_three = getByLabelText("Choice Three") as HTMLInputElement;
+
+		expect(item_one).toBeChecked();
+		expect(item_two).not.toBeChecked();
+		expect(item_three).toBeChecked();
+
+		expect(component.value).toEqual([1, 3]);
+	});
+
+	test("changing the component value updates the checkboxes", async () => {
+		const { getByLabelText, debug, component } = await render(CheckboxGroup, {
+			value: [],
+			label: "Dropdown",
+			choices: [
+				["Choice One", 1],
+				["Choice Two", 2],
+				["Choice Three", 3]
+			]
+		});
+
+		const item_one = getByLabelText("Choice One") as HTMLInputElement;
+		const item_two = getByLabelText("Choice Two") as HTMLInputElement;
+		const item_three = getByLabelText("Choice Three") as HTMLInputElement;
+
+		expect(item_one).not.toBeChecked();
+		expect(item_two).not.toBeChecked();
+		expect(item_three).not.toBeChecked();
+
+		component.value = [1, 3];
+
+		expect(item_one).toBeChecked();
+		expect(item_two).not.toBeChecked();
+		expect(item_three).toBeChecked();
+	});
+
+	test("setting a value that does not exist does nothing", async () => {
+		const { getByLabelText, component } = await render(CheckboxGroup, {
+			value: [],
+			label: "Dropdown",
+			choices: [
+				["Choice One", 1],
+				["Choice Two", 2],
+				["Choice Three", 3]
+			]
+		});
+
+		const item_one = getByLabelText("Choice One") as HTMLInputElement;
+		const item_two = getByLabelText("Choice Two") as HTMLInputElement;
+		const item_three = getByLabelText("Choice Three") as HTMLInputElement;
+
+		expect(item_one).not.toBeChecked();
+		expect(item_two).not.toBeChecked();
+		expect(item_three).not.toBeChecked();
+
+		component.value = ["choice_one"];
+
+		expect(item_one).not.toBeChecked();
+		expect(item_two).not.toBeChecked();
+		expect(item_three).not.toBeChecked();
+	});
+});
+
+describe("Events", () => {
```

Comment:
Events are the other important thing in this case. We want to test a few different cases again. We have three events to think about here and their behaviour is slightly different. We want to test that the fire when they should but we also want to ensure they do not fire when they shouldn't too. We also need to check payloads in certain cases (`select`).

We have a `listen` utility to help with this (and it is fully typechecked so you should get autocomplete on these event names).